test/functional/setup.ts

```diff
@@ -54,6 +64,11 @@ class FunctionalTestSetup {
     this.mochaContext.timeout(0)
     return queryExecutionDebugger
   }
+
+  public addScraper(instructionsArg: string, folder: string, options?: Options) {
+    const newSize = this.scrapers.push(new ScraperProgram(instructionsArg, folder, options))
+    return this.scrapers[newSize - 1]
+  }
```

Comment:
so we can do a little better here:
```suggestion
  public attachScraper(...args: ConstructorParameters<typeof ScraperProgram>) {
    const scraper = new ScraperProgram(...args)
    this.scrapers.push(scraper)
    return scraper
  }
```

Reply:
I love the idea of using `ConstructorParameters`, but after trying, it doesn't look like it works with our overloaded constructor. [Here](https://github.com/microsoft/TypeScript/issues/37079) is an issue that seems to showcase the same behavior

Reply:
disappointing, but that fix looks a little too hacky, lets just stick with what you have